http2-flowcontrol-sim/src/Simulation.java

```diff
@@ -0,0 +1,94 @@
+import java.util.ArrayList;
+
+public class Simulation {
+  private int howLong;
+  private SimulationClock clock;
+  private ArrayList<Connection> connections;
+
+  /**
+   * Return a new simulation with the requested number of streams and connections all listening to
+   * the same simulation clock. Parameters window, latency, and bandwidth are used to set the
+   * initial window size, latency and bandwidth of new connections. The clock runs for 'howLong'
+   * ticks.
+   *
+   * @param window
+   * @param latency
+   * @param bandwidth
+   * @param numConnections
+   * @param numStreams
+   * @param howLong
+   */
+  public Simulation(int window, int latency, int bandwidth, int numConnections, int numStreams,
+      int howLong) {
+    this.howLong = howLong;
+    clock = new SimulationClock();
+    connections = new ArrayList<Connection>();
+    for (int i = 0; i < numConnections; i++) {
+      Connection c = new Connection(clock, window, latency, bandwidth);
+      for (int j = 0; j < numStreams; j++) {
```

Comment:
personal preference:  I find the use of "j" to be confusing, since it looks so similar to i.  You might consider using k as your inner iterator.